diff --git a/src/lib/Hydra/Controller/Build.pm b/src/lib/Hydra/Controller/Build.pm
index 2571d11..1a1b466 100644
--- a/src/lib/Hydra/Controller/Build.pm
+++ b/src/lib/Hydra/Controller/Build.pm
@@ -171,7 +171,7 @@ sub showLog {
     }
 
     elsif ($mode eq "tail-reload") {
-        my $url = $c->uri_for($c->request->uri->path);
+        my $url = $c->request->uri->as_string;
         $url =~ s/tail-reload/tail/g;
         $c->stash->{url} = $url;
         $c->stash->{reload} = !$c->stash->{build}->finished;
