diff --git a/nix.patch b/nix.patch
new file mode 100644
index 0000000..2969b85
--- /dev/null
+++ b/nix.patch
@@ -0,0 +1,13 @@
+diff --git a/src/libstore/local-fs-store.cc b/src/libstore/local-fs-store.cc
+index 7094a50..f8249d9 100644
+--- a/src/libstore/local-fs-store.cc
++++ b/src/libstore/local-fs-store.cc
+@@ -22,7 +22,7 @@ struct LocalStoreAccessor : public FSAccessor
+ 
+         struct stat st;
+         if (lstat(path.c_str(), &st)) {
+-            if (errno == ENOENT) return {Type::tMissing, 0, false};
++            if (errno == ENOENT || errno == ENOTDIR) return {Type::tMissing, 0, false};
+             throw SysError(format("getting status of ‘%1%’") % path);
+         }
+ 
diff --git a/release.nix b/release.nix
index e482b2a..1fcf3ce 100644
--- a/release.nix
+++ b/release.nix
@@ -83,7 +83,15 @@ rec {
 
     let
 
-      nix = nixUnstable;
+      nix = overrideDerivation nixUnstable (drv: rec {
+        name = "nix-1.12pre4499_b4e0335";
+        src = fetchurl {
+          url = "https://hydra.nixos.org/build/32694605/download/4/"
+              + "${name}.tar.xz";
+          sha256 = "15lmpyn0m2qnrpcwd6g9nkssps6qfzrmcsf534xb41kw9mh42frb";
+        };
+        patches = (drv.patches or []) ++ [ ./nix.patch ];
+      });
 
       NetStatsd = buildPerlPackage {
         name = "Net-Statsd-0.11";
diff --git a/src/lib/Hydra/Controller/Build.pm b/src/lib/Hydra/Controller/Build.pm
index 2571d11..1a1b466 100644
--- a/src/lib/Hydra/Controller/Build.pm
+++ b/src/lib/Hydra/Controller/Build.pm
@@ -171,7 +171,7 @@ sub showLog {
     }
 
     elsif ($mode eq "tail-reload") {
-        my $url = $c->uri_for($c->request->uri->path);
+        my $url = $c->request->uri->as_string;
         $url =~ s/tail-reload/tail/g;
         $c->stash->{url} = $url;
         $c->stash->{reload} = !$c->stash->{build}->finished;
