diff --git a/apps/ejabberd/src/ejabberd_s2s_out.erl b/apps/ejabberd/src/ejabberd_s2s_out.erl
index 886fe088..04a7f5e0 100644
--- a/apps/ejabberd/src/ejabberd_s2s_out.erl
+++ b/apps/ejabberd/src/ejabberd_s2s_out.erl
@@ -657,7 +657,7 @@ wait_for_starttls_proceed({xmlstreamelement, El}, StateData) ->
 		    Socket = StateData#state.socket,
 		    TLSOpts = case ejabberd_config:get_local_option(
 				     {domain_certfile,
-				      binary_to_list(StateData#state.myname)}) of
+				      StateData#state.myname}) of
 				  undefined ->
 				      StateData#state.tls_options;
 				  CertFile ->
