diff --git a/apps/ejabberd/src/ejabberd_loglevel.erl b/apps/ejabberd/src/ejabberd_loglevel.erl
index b8bf8cc..af2b810 100644
--- a/apps/ejabberd/src/ejabberd_loglevel.erl
+++ b/apps/ejabberd/src/ejabberd_loglevel.erl
@@ -68,9 +68,7 @@ set(Level) when is_integer(Level) ->
     {_, Name} = lists:keyfind(Level, 1, ?LOG_LEVELS),
     set(Name);
 set(Level) ->
-    Path = log_path(),
-    lager:set_loglevel(lager_console_backend, Level),
-    lager:set_loglevel(lager_file_backend, Path, Level).
+    lager:set_loglevel(lager_console_backend, Level).
 
 set_custom(Module, Level) when is_integer(Level) ->
     {_, Name} = lists:keyfind(Level, 1, ?LOG_LEVELS),
diff --git a/rel/files/app.config b/rel/files/app.config
index 33fec09..fdbe4a1 100644
--- a/rel/files/app.config
+++ b/rel/files/app.config
@@ -1,10 +1,9 @@
 [
 {lager, [
     {handlers, [
-        {lager_console_backend, [info, {lager_default_formatter,[{eol, "\r\n"}]}]},
-        %% file is shadowed by ejabberd.cfg
-        {lager_file_backend, [{file, "log/ejabberd.log"}, {level, info}, {size, 2097152}, {date, "$D0"}, {count, 5}]}
-    ]}
+        {lager_console_backend, [info, {lager_default_formatter,[{eol, "\n"}]}]}
+    ]},
+    {crash_log, undefined}
   ]}
 ].
 
