diff --git a/apps/ejabberd/src/ejabberd_loglevel.erl b/apps/ejabberd/src/ejabberd_loglevel.erl
index de61348..d121232 100644
--- a/apps/ejabberd/src/ejabberd_loglevel.erl
+++ b/apps/ejabberd/src/ejabberd_loglevel.erl
@@ -58,8 +58,7 @@ set(Level) when is_integer(Level) ->
     {_, Name} = lists:keyfind(Level, 1, ?LOG_LEVELS),
     set(Name);
 set(Level) ->
-    ok = lager:set_loglevel(lager_console_backend, Level),
-    ok = lager:set_loglevel(lager_file_backend, ?LOG_PATH, Level).
+    ok = lager:set_loglevel(lager_console_backend, Level).
 
 set_custom(Module, Level) when is_integer(Level) ->
     {_, Name} = lists:keyfind(Level, 1, ?LOG_LEVELS),
diff --git a/rel/files/app.config b/rel/files/app.config
index 96783c3..af192c5 100644
--- a/rel/files/app.config
+++ b/rel/files/app.config
@@ -1,9 +1,9 @@
 [
 {lager, [
     {handlers, [
-        {lager_console_backend, [info, {lager_default_formatter,[{eol, "\r\n"}]}]},
-        {lager_file_backend, [{file, "log/ejabberd.log"}, {level, info}, {size, 2097152}, {date, "$D0"}, {count, 5}]}
-    ]}
+        {lager_console_backend, [info, {lager_default_formatter,[{eol, "\n"}]}]}
+    ]},
+    {crash_log, undefined}
   ]}
 ].
 
