diff --git a/apps/ejabberd/src/mod_pubsub.erl b/apps/ejabberd/src/mod_pubsub.erl
index c139ee93..4239f17d 100644
--- a/apps/ejabberd/src/mod_pubsub.erl
+++ b/apps/ejabberd/src/mod_pubsub.erl
@@ -90,7 +90,7 @@
 %% API and gen_server callbacks
 -export([start_link/2, start/2, stop/1, init/1,
          handle_call/3, handle_cast/2, handle_info/2,
-         terminate/2, code_change/3]).
+         terminate/2, code_change/3, deps/2]).
 -export([default_host/0]).
 
 -export([send_loop/1]).
@@ -4268,3 +4268,7 @@ timestamp() ->
 
 db_type(_Host) ->
     mnesia.
+
+deps(_Host, Opts) ->
+    [{mod_caps, hard} || P <- gen_mod:get_opt(plugins, Opts, []),
+                         P == <<"pep">>].
