diff --git a/rel/reltool.config.script b/rel/reltool.config.script
index 40877c9..bfc9cbf 100644
--- a/rel/reltool.config.script
+++ b/rel/reltool.config.script
@@ -55,7 +55,7 @@ IncludeApps = lists:map(fun(App) -> {app, App, [{incl_cond, include}]} end, Apps
 
 
 [{sys, [
-        {lib_dirs, ["../apps", "../deps"]},
+        {lib_dirs, ["../apps"]},
         {incl_cond, exclude},
         {rel, "mongooseim", "", [mongoose | AppsToRun]},
         {rel, "start_clean", "", [kernel,stdlib]},
@@ -64,6 +64,7 @@ IncludeApps = lists:map(fun(App) -> {app, App, [{incl_cond, include}]} end, Apps
         {excl_archive_filters, [".*"]}, % do no archive built libs
         {excl_sys_filters, ["^bin/.*",
                             "^erts.*/bin/(dialyzer|typer)"]},
+        {excl_lib, otp_root},
 
         {app, mongoose, [{incl_cond, include}, {lib_dir, ".."}]}
        ] ++ IncludeApps},
@@ -80,8 +81,6 @@ IncludeApps = lists:map(fun(App) -> {app, App, [{incl_cond, include}]} end, Apps
            {copy, "files/mongooseimctl", "bin/mongooseimctl"},
            {copy, "files/app.config", "etc/app.config"},
            {template, "files/vm.args", "etc/vm.args"},
-           {template, "files/ejabberd.cfg", "etc/ejabberd.cfg"},
-           {mkdir, "var"},
-           {mkdir, "var/log"}
+           {template, "files/ejabberd.cfg", "etc/ejabberd.cfg"}
           ]}
 ].
