diff --git a/apps/ejabberd/src/ejabberd_app.erl b/apps/ejabberd/src/ejabberd_app.erl
index 634a1c5..50bf928 100644
--- a/apps/ejabberd/src/ejabberd_app.erl
+++ b/apps/ejabberd/src/ejabberd_app.erl
@@ -68,7 +68,11 @@ start(normal, _Args) ->
     mongoose_metrics:init(),
     ejabberd_listener:start_listeners(),
     ejabberd_admin:start(),
-    ?INFO_MSG("ejabberd ~s is started in the node ~p", [?VERSION, node()]),
+    Message = io_lib:format(
+        "MongooseIM ~s has finished starting up on node ~p",
+        [?VERSION, node()]
+    ),
+    sd_notify:sd_notify(0, "READY=1\nSTATUS=" ++ Message ++ "\n"),
     Sup;
 start(_, _) ->
     {error, badarg}.
diff --git a/rebar.config b/rebar.config
index c19db14..9937c04 100644
--- a/rebar.config
+++ b/rebar.config
@@ -39,7 +39,8 @@
   {usec, ".*", {git, "git://github.com/esl/usec.git", {branch, "master"}}},
   {ecoveralls, ".*", {git, "git://github.com/nifoc/ecoveralls.git", "40fa0d2f2057fff29e964f94fccf6ef2f13d34d2"}},
   {mustache, ".*", {git, "git://github.com/mojombo/mustache.erl.git", "d0246fe143058b6404f66cf99fece3ff6e87b7ed"}},
-  {recon, "2.2.1", {git, "git://github.com/ferd/recon.git", {tag, "2.2.1"}}}
+  {recon, "2.2.1", {git, "git://github.com/ferd/recon.git", {tag, "2.2.1"}}},
+  {sd_notify, "1", {git, "git://github.com/systemd/erlang-sd_notify.git", {branch, "master"}}}
  ]}.
 
 {pre_hooks, [{compile, "tools/compile_riak_pb.sh"}]}.
diff --git a/rel/reltool.config.script b/rel/reltool.config.script
index 3f9b26e..4ca657c 100644
--- a/rel/reltool.config.script
+++ b/rel/reltool.config.script
@@ -45,7 +45,8 @@ BaseAppsToInclude = AppsToRun ++
                      alarms,
                      idna,
                      recon,
-                     setup
+                     setup,
+                     sd_notify
                      ],
 
 
