diff --git a/apps/ejabberd/src/ejabberd_app.erl b/apps/ejabberd/src/ejabberd_app.erl
index da18651..5cd6107 100644
--- a/apps/ejabberd/src/ejabberd_app.erl
+++ b/apps/ejabberd/src/ejabberd_app.erl
@@ -68,7 +68,17 @@ start(normal, _Args) ->
     start_modules(),
     ejabberd_listener:start_listeners(),
     ejabberd_admin:start(),
-    ?INFO_MSG("ejabberd ~s is started in the node ~p", [?VERSION, node()]),
+
+    %% Notify systemd about startup
+    Message = io_lib:format("ejabberd ~s is started on node ~p",
+                            [?VERSION, node()]),
+    EscMessage = lists:flatmap(fun($') -> [$', $\\, $', $'];
+                                  (C)  -> [C] end, Message),
+    SdNotify = "systemd-notify --ready --status='" ++ EscMessage ++ "'",
+    % XXX: Very ugly workaround because of the following issue:
+    %      http://comments.gmane.org/gmane.comp.sysutils.systemd.devel/18649
+    timer:apply_after(10000, os, cmd, [SdNotify]),
+
     Sup;
 start(_, _) ->
     {error, badarg}.
