diff --git a/tests/ejabberdctl_SUITE.erl b/tests/ejabberdctl_SUITE.erl
index 79fbca8..0f9f093 100644
--- a/tests/ejabberdctl_SUITE.erl
+++ b/tests/ejabberdctl_SUITE.erl
@@ -80,20 +80,15 @@ suite() ->
     escalus:suite().
 
 init_per_suite(Config) ->
-    {ok, EjdWD} = escalus_ejabberd:rpc(file, get_cwd, []),
     Cwd0 = escalus_config:get_config(data_dir, Config),
     CwdTokens = string:tokens(Cwd0, "/"),
     Cwd =  [$/ | string:join(lists:sublist(CwdTokens, 1, length(CwdTokens)-2), "/")],
     TemplatePath = Cwd ++ "/roster.template",
     start_mod_admin_extra(),
-    CtlPath = case filelib:is_file(EjdWD ++ "/bin/ejabberdctl") of
-                  true -> EjdWD ++ "/bin/ejabberdctl";
-                  false -> EjdWD ++ "/bin/mongooseimctl"
-              end,
 
     AuthMods = auth_modules(),
     
-    NewConfig = escalus:init_per_suite([{ctl_path, CtlPath},
+    NewConfig = escalus:init_per_suite([{ctl_path, os:find_executable("mongooseimctl")},
                                         {ctl_auth_mods, AuthMods},
                                         {roster_template, TemplatePath} | Config]),
     escalus:create_users(NewConfig, {by_name, [alice, mike, bob, kate]}).
@@ -621,7 +616,7 @@ get_user_data(User, _Config) ->
     {Username, Domain, Password}.
 
 run(Cmd) -> 
-    run(Cmd, 5000).
+    run(Cmd, 60000).
 
 run(Cmd, Timeout) ->
     Port = erlang:open_port({spawn, Cmd},[exit_status]),
